gemrb/core/Scriptable/Actor.cpp

```diff
@@ -7457,7 +7444,17 @@ void Actor::ApplyModal(const ResRef& modalSpell)
 {
 	unsigned int aoe = ModalStates[Modal.State].aoe_spell;
 	if (aoe == 1) {
-		core->ApplySpellPoint(modalSpell, GetCurrentArea(), Pos, this, 0);
+		if (Modal.State == MS_TURNUNDEAD) {
+			if (!area) return;
+			std::vector<Actor*> targets = area->GetAllActorsInRadius(Pos, GA_NO_LOS|GA_NO_DEAD|GA_NO_NEUTRAL|GA_NO_ALLY|GA_NO_SELF|GA_NO_UNSCHEDULED, GetSafeStat(IE_VISUALRANGE));
+			for (const auto& target : targets) {
+				if (GameScript::ID_Alignment(this, AL_EVIL) || target->GetStat(IE_GENERAL) == GEN_UNDEAD) {
+					core->ApplySpell(modalSpell, target, this, 0);
+				}
+			}
+		} else {
+			core->ApplySpellPoint(modalSpell, GetCurrentArea(), Pos, this, 0);
+		}
```

Comment:
Why is this needed at all? This can all be done in the spell itself (and should be already). However it's not clear if turning paladins still works properly under this change.

Reply:
> Why is this needed at all? This can all be done in the spell itself (and should be already). However it's not clear if turning paladins still works properly under this change.

As I already said, it cast turn undead at the caster himself. And this is not an effect with a sub-effect. The paladin made save throw from turn undead.

Reply:
That's not necessarily bad if it's an aoe (which this is, it's using inareans.pro that ignores the center). But the code you removed was not for when paladins are turning, but when evil clerics are turning — they can turn paladins as well, besides the undead.

Reply:
The new code should also work on evil clerics.

Reply:
Not really, you explicitly exclude almost all good creatures (GA_NO_ALLY) when iterating over potential targets.

Reply:
Done fix it

Reply:
Like I said at the start, I don't believe this whole chunk is needed at all (and what's up with the random alignment check?). What's actually broken now?

Reply:
Take a paladin and try to use turn undead. By the way, why is he from level 5 if he should be from level 3? The paladin will roll a saving throw against his own turn undead and the surrounding undead will not be affected. This effect has no sub-effect.

Reply:
I tried both a cleric and a paladin and both turned skeletons fine. It was ToB, so the level difference was enough that when they were affected (only one managed a save once in about 10 rolls), they got destroyed. Also means high saves on the paladin already. Maybe you were unlucky. Maybe something has been fixed since. Either way I don't see anything needing fixing here.